src/main/java/com/uipath/uipathpackage/UiPathDeploy.java

```diff
@@ -106,37 +108,46 @@ public BuildStepMonitor getRequiredMonitorService() {
     /**
      * Run this step.
      *
-     * @param run      a build this is running as a part of
-     * @param filePath a workspace to use for any file operations
-     * @param launcher a way to start processes
-     * @param listener a place to send output
+     * @param run       a build this is running as a part of
+     * @param workspace a workspace to use for any file operations
+     * @param launcher  a way to start processes
+     * @param listener  a place to send output
      * @throws InterruptedException if the step is interrupted
      * @throws IOException          if something goes wrong
      */
     @Override
-    public void perform(@Nonnull Run<?, ?> run, @Nonnull FilePath filePath, @Nonnull Launcher launcher, @Nonnull TaskListener listener) throws InterruptedException, IOException {
-        EnvVars env = run.getEnvironment(listener);
-        ResourceBundle rb = ResourceBundle.getBundle("config");
+    public void perform(@Nonnull Run<?, ?> run, @Nonnull FilePath workspace, @Nonnull Launcher launcher, @Nonnull TaskListener listener) throws InterruptedException, IOException {
         Utility util = new Utility();
         util.validateParams(orchestratorAddress, "Invalid Orchestrator Address");
         util.validateParams(packagePath, "Invalid Package Path");
-        String packagePathFormatted = escapePowerShellString(env.expand(packagePath.trim()));
-        String orchestratorTenantFormatted = env.expand(orchestratorTenant.trim()).isEmpty() ? util.getValue(rb, "UiPath.DefaultTenant") : escapePowerShellString(env.expand(orchestratorTenant.trim()));
-        StandardUsernamePasswordCredentials cred = CredentialsProvider.findCredentialById(credentialsId, StandardUsernamePasswordCredentials.class, run, Collections.emptyList());
-        if (cred == null || cred.getUsername().isEmpty() || cred.getPassword().getPlainText().isEmpty()) {
-            throw new AbortException("Invalid credentials");
-        }
-        String username = escapePowerShellString(cred.getUsername()); //not using escapeJava as it will corrupt the username and password
-        String password = escapePowerShellString(cred.getPassword().getPlainText());
-        listener.getLogger().println("Opening Powershell Session");
+        FilePath tempRemoteDir = tempDir(workspace);
         try {
```

Comment:
Put the try before the above statement. You may run into authorization issues creating tempDir

Reply:
This is not yet fixed